src/PrivateRoute.js

```diff
@@ -0,0 +1,10 @@
+import { Navigate, Outlet } from "react-router-dom";
+import { useAuth } from "./context/auth-context";
+
+const PrivateRoute = () => {
+  const { authState } = useAuth();
+
+  return authState.user ? <Outlet /> : <Navigate to="/signin" replace />;
```

Comment:
add the location as the user needs to programmatically redirect to the page which comes in the private routes

Reply:
Done the changes